django/db/models/fields/__init__.py

```diff
@@ -373,6 +381,25 @@ def _check_choices(self):
             )
         ]
 
+    def _check_db_default(self):
+        if self.db_default is NOT_PROVIDED:
+            return []
+
+        if (
+            hasattr(self, "model")
+            and "supports_expression_defaults" in self.model._meta.required_db_features
+        ):
+            # If the database does not support functions in defaults
+            # and the model is explicitly marked as requiring that feature,
+            # blocking management commands by returning an Error is unhelpful.
+            return []
+
+        try:
+            connection.ops.check_field_default_support(self.db_default)
+        except NotSupportedError as error:
+            return [checks.Error(error.args[0], obj=self, id="fields.E011")]
+        return []
```

Comment:
We're using the exception to pass a custom message from the backend to the check. It's not as simple as just returning a boolean. Perhaps we could return `None` or the message?

Reply:
I would raise here an error for all database backends that don't support (`supports_expression_defaults`) and add `unsupported_default_expression` to the database features, instead of using `check_field_default_support()` and re-raise errors, e.g.:
```python
    def _check_db_default(self, databases=None, **kwargs):
        if self.db_default is NOT_PROVIDED:
            return []
        errors = []
        for db in databases:
            if not router.allow_migrate_model(db, self.model):
                continue
            connection = connections[db]
            if not (
                connection.features.supports_expression_defaults
                or "supports_expression_defaults" in self.model._meta.required_db_features
            ):
                 errors.append(checks.Error(...))
            if (
                (
                    isinstance(expression, connection.features.unsupported_default_expression)
                    or not expression.allowed_default
                )
                and "supports_expression_defaults" not in self.model._meta.required_db_features
            ):
                errors.append(checks.Error(...))
        return errors
``` 

Also, this check must rely on `databases`.

Reply:
I think I've made the `databases` changes we need here. I'm not clear what you're aiming for with `unsupported_default_expression` though.

Reply:
What would the value of `unsupported_default_expression` be for `MySQL`? And what should it default to?

Are you suggesting I hardcode the check for `Now` and `Value` in `_check_db_default`? Or where should that live?

Reply:
Sorry, I even confused myself :smile: The current form looks good :+1: